sklearn/model_selection/tests/test_validation.py

```diff
@@ -1184,6 +1184,73 @@ def test_learning_curve_with_shuffle():
 
 
+def test_learning_curve_with_stratify():
+    def max_diff_in_ones_between_strata(X, y, stratify, folds_count=5,
+                                        iters_count=10, random_state=0):
+        ones_counter = []
+
+        class MockCountingEstimator(BaseEstimator):
+            """ Dummy classifier: it stores (in global var for this class)
+            count of non-zero-labeled elements in passed y """
+            def fit(self, X_subset, y_subset=None):
+                cur_ones_count = np.count_nonzero(y_subset)
+                ones_counter.append(cur_ones_count)
+                return self
+
+            def score(self, X, y):
+                return 0
+
+        counting_estimator = MockCountingEstimator()
+        cv = StratifiedKFold(folds_count)
+        train_sizes = np.linspace(1.0 / iters_count, 1.0,
+                                  iters_count)
+
+        learning_curve(counting_estimator, X, y, cv=cv,
+                       train_sizes=train_sizes, shuffle=True,
+                       stratify=stratify,
+                       random_state=random_state)
+
+        # Extract max possible diff of ones count between strata
+        # inside each training CV fold
+        ones_counter = np.asarray(ones_counter).reshape(folds_count,
+                                                        iters_count)
+        # ones_counter stores accumulated count, not count in each stratum,
+        # so we need to take this into account: e.g. [1 1 6 8 9] => [1 0 5 2 1]
+        strata_ones = np.hstack((ones_counter[:, [0]], np.diff(ones_counter)))
+        max_diff = np.max(np.max(strata_ones, 1) - np.min(strata_ones, 1))
+        return max_diff
+
+    ones = 25
+    zeros = 2475
+    X = np.random.rand(ones + zeros, 2)
+    y = np.array([1] * ones + [0] * zeros)
+    cv = StratifiedKFold(5)
+
+    # Check that without stratify it is possible to have training sets
+    # with just one classes for some random seeds
+    estimator = PassiveAggressiveClassifier(max_iter=5, tol=None,
+                                            shuffle=False)
+
+    # e.g. for this data without stratify it works only with random_state=2,
+    # but not 0 or 1
+    learning_curve(estimator, X, y, cv=cv, train_sizes=[0.1], shuffle=True,
+                   random_state=2)
+    assert_raises(ValueError, learning_curve, estimator, X, y, cv=cv,
+                  train_sizes=[0.1], shuffle=True, random_state=0)
+    assert_raises(ValueError, learning_curve, estimator, X, y, cv=cv,
+                  train_sizes=[0.1], shuffle=True, random_state=1)
+
+    # ...but with stratify=True it works for all random seeds
+    for rand_state in range(10):
+        learning_curve(estimator, X, y, cv=cv, train_sizes=[0.1], shuffle=True,
+                       stratify=True, random_state=rand_state)
+
+    # Check that stratify indeed makes strata with almost equal counts of ones
+    # (i.e. differing by not more than 2)
```

Comment:
1?

Reply:
Yep, this is one of 2 hard problems in computer science.